internal/cli/certificate.go

```diff
@@ -278,9 +279,14 @@ func createCsr(o *CertificateOptions, name string, priv crypto.PrivateKey) ([]by
 		return nil, err
 	}
 
-	// the CN is going to be a FC-generated UUID, populated at signing time
+	if name == "" {
+		name = uuid.NewString()
+	}
```

Comment:
Also, we should do this earlier - maybe as the first step in `Run()` [here](https://github.com/flightctl/flightctl/blob/main/internal/cli/certificate.go#L142) - and populate this into `o.Name` to make sure that the files written out for the user correspond to this CN as well.

Reply:
That's okay, I can also do it as a follow-on PR if you want.

Reply:
Let's do it this way.